### ISSUE TITLE
Suppress recommendations for conflicting extensions

### ISSUE THREAD

**Issue by sam-mccall** [microsoft/vscode#77979]
**TL:DR**; once the user has installed an extension that conflicts with a recommended one, it'd be helpful to stop recommending it.

Hi! As part of LLVM, I work on the `clangd` C++ language server, which has a [vscode plugin](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd).

Microsoft also has the [vscode-cpptools](https://github.com/microsoft/vscode-cpptools) language server for C++.

We've noticed that the recommended extension system makes it hard to get clangd set up and keep it working. I'm sure this wasn't the intent - it's a great system for getting new users set up with working tools. However, there's a few interacting issues here here:
 - the `vscode-cpptools` and `clangd` extensions conflict in various ways, and can't be used together
 - the `vscode-cpptools` extension is recommended to anyone opening a c++ file (fair enough)
 - the `vscode-cpptools` extension is called only "C/C++" in the recommendation dialog, which is confusing to users that want C/C++ support from a different language server
 - the `vscode-cpptools` extension continues to be recommended when opening C++ files, even once `vscode-clangd` is installed.

In practice, we've found that a very large fraction of new VSCode users specifically trying to install clangd end up with both extensions installed and active eventually, and we need to include explicit instructions about how to remove this plugin.

So a couple of suggestions, which I don't think will hurt the new-user path or adoption of recommended extensions:
 - Allow extension metadata to include a list of UUIDs for conflicting extensions (like the opposite of `extensionDependencies`). This would suppress recommendation of that extension, and potentially allow VSCode to flag conflicts between installed extensions to the user.
 - Include the publisher name when prompting to install a recommended extension. e.g. `The 'C/C++' extension from Microsoft is recommended for this file type`. This makes the scope clearer to the user I think.

(#77957 will improve this experience a little for users who understand what's going on, but I don't think is sufficient.)

**Comment by sandy081** [comment]
@chrisdias Assigning to you since it is related to product recommendations. 

May be do we need to add this extension also to recommendations?

**Comment by chrisdias** [comment]
i'm hesitant to add `vscode-clangd` to the list of recommendations because of this comment in the `readme`:

> Note: clangd is under heavy development, not all LSP features are implemented. See Current Status for details.

**Comment by chrisdias** [comment]
Related, https://github.com/microsoft/vscode/issues/41368 has conflicting "important" suggestions - we can't tell the difference between mysql and sql server based on the `.sql` file extension.  In that case, I added some recommendations (passive) for extensions that work with mysql to try to strike a balance.  I could do the same here, add `vscode-clangd` but that doesn't address the conflicts between two extensions.

**Comment by sam-mccall** [comment]
Thanks for looking at adding recommendations! The "clangd is under heavy development" and other text is pretty old, dating back to the prototype. I'll take a pass at updating the text/metadata in the next couple of days, if that will help.
(Our work has been almost exclusively on the language server, and the extension mostly just activates the languageclient library).

That said, the recommendations are really up to you of course - my primary request is to reduce the confusion *after* clangd is installed. I'd think that adding an `extensionConflicts` or similar to `package.json` might be useful for other things too (e.g. warning users who are trying to install conflicting extensions).

**Comment by isidorn** [comment]
@sam-mccall is this request still valid? If yes, do let us know and there might be an option we can do now.
If not do let us know and I will close this issue. Thanks!

**Comment by sam-mccall** [comment]
@isidorn I'm sorry, i missed the notification in August. This is still relevant to us if there's anything that can be done!

**Comment by isidorn** [comment]
@sam-mccall thanks for letting us know. Let me reopen and assign it to the current milestone to investigate if this is something we can improve.

**Comment by isidorn** [comment]
We have pushed a fix for this and C++ will no longer be recommended when clangd is installed.
You can try it out with VS Code insiders from next week. Thanks
